src/search.jl

```diff
@@ -5,6 +5,7 @@ function search!(t::Tree, traversal_func::Function, x::AbstractVector,
     std::AbstractVector, prunable;
     maxiter::Int = 32, regularization::Bool = true, tol::Real = DEFAULT_TOL)
 
+    # TODO: pre-allocate Vector{<:Node}(undef, ...) to avoid thread race condition
     resulting_nodes = Node[]
```

Comment:
@MingChiangChang Could you have a look at this and see if you can pre-allocate `resulting_nodes` and index into it instead of using `push!`, and similarly for `deleting`? Alternatively, we'd need to put locks around the `push!` calls to avoid two threads pushing to the same location at the same time. I got an error locally because of this.

Reply:
Yes, I can do that